engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -217,6 +217,7 @@
   protected String shutdownDiagnosticsMessage = "";
   private long lastResourceRequest = 0;
   private final Map<String, StreamingContainerAgent> containers = new ConcurrentHashMap<>();
+  private final Set<String> containerHeartBeatMissSet = Collections.newSetFromMap(new ConcurrentHashMap<String, Boolean>());
```

Comment:
Yes it will be easier, but when I was testing it I observed multiple container error events for the heartbeat miss of the same container , although this wasn't wrong but according to me only one container error event should be raised for this scenario and hence I have used a Set to assure if the container has already raised this error event it should not raise it again.